File: problems/442_find_all_duplicates_in_an_array.py
class Solution:
    def findDuplicates(self, nums: list[int]) -> list[int]:
        dups: list[int] = []
        for num in nums:
            index = abs(num) - 1
            index_is_negative = nums[index] < 0
            if index_is_negative:
                dups.append(abs(num))
            else:
                nums[index] *= -1
        return dups

File: problems/442-find-all-duplicates-in-an-array.ts
function findDuplicates(nums: number[]): number[] {
  let duplicates: number[] = []
  for (const num of nums) {
    const index = Math.abs(num) - 1
    const indexIsNegative = nums[index] < 0
    if (indexIsNegative) {
      duplicates.push(Math.abs(num))
    } else {
      nums[index] *= -1
    }
  }
  return duplicates
}
